version: '3.8'

services:
  postgres:
    image: postgres:15-alpine
    container_name: postgres
    environment:
      POSTGRES_DB: UpstartDb
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - app-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5

  flyway:
    image: flyway/flyway:latest
    command: -configFiles=/flyway/conf/flyway-docker.conf migrate
    volumes:
      - ./flyway/upstart:/flyway/conf
      - ./flyway/upstart/migrations:/flyway/sql
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - app-network

  seq:
    image: datalust/seq:latest
    container_name: seq
    environment:
      - ACCEPT_EULA=Y
      - SEQ_FIRSTRUN_NOAUTHENTICATION=true
    ports:
      - "5341:80"
    volumes:
      - seq_data:/data
    networks:
      - app-network
    restart: unless-stopped



volumes:
  postgres_data:
  seq_data:

networks:
  app-network:
    driver: bridge